shared/src/commonMain/kotlin/world/respect/domain/opds/validator/OpdsValidator.kt

```diff
@@ -0,0 +1,601 @@
+package world.respect.domain.opds.validator
+
+import world.respect.domain.opds.model.OpdsFeed
+import world.respect.domain.opds.model.OpdsFeedMetadata
+import world.respect.domain.opds.model.OpdsPublication
+import kotlinx.serialization.json.Json
+import kotlinx.serialization.json.JsonObject
+import kotlinx.serialization.SerializationException
+import java.io.File
+import java.net.URI
+import java.net.URL
+
+/**
+ * Validates OPDS feeds and publications according to the OPDS 2.0 specification.
+ * In RESPECT context, OPDS feeds are used to list learning units, where each
+ * publication represents a distinct learning unit.
+ * For reference, see the schemas:
+ * - https://drafts.opds.io/schema/feed.schema.json
+ * - https://drafts.opds.io/schema/publication.schema.json
+ */
+class OpdsValidator {
+
+    // List to store warnings that don't cause validation failure
+    private val warnings = mutableListOf<String>()
```

Comment:
Clearly and obviously wrong: why is this a member variable on the class? The list of warnings needs built up per validation run.